ignite/metrics/ssim.py

```diff
@@ -157,18 +159,34 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
                 f"Expected y_pred and y to have BxCxHxW shape. Got y_pred: {y_pred.shape} and y: {y.shape}."
             )
 
-        channel = y_pred.size(1)
-        if len(self._kernel.shape) < 4:
-            self._kernel = self._kernel.expand(channel, 1, -1, -1).to(device=y_pred.device)
+        if y_pred.device != self._kernel.device:
+            if self._kernel.device == torch.device("cpu"):
+                self._kernel = self._kernel.to(device=y_pred.device)
+                self._sum_of_ssim = self._sum_of_ssim.to(device=y_pred.device)
```

Comment:
I do not think that we should move _sum_of_ssim to another device. It should always stay on self._device

Reply:
What would be the benefit of keeping the sum on self._device ? That would mean moving the current value on the right device on each update call, just to spare 64 bits on GPU.

"Because the user wants it" is a valid answer :)

Reply:
> What would be the benefit of keeping the sum on self._device ?

It is a feature we have for all metrics that user can choose where to store internal intermediate parameters.
So, yes, user picks it.